src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -226,128 +185,32 @@ && localCheckoutProvided()
-                                coreCoordinates.version, plugin.getDisplayName(), plugin.version),
-                        t);
-                errorMessage = t.getMessage();
-                pomData = null;
-            }
-
-            Set<String> testDetails = new TreeSet<>();
-            if (errorMessage == null) {
-                try {
-                    TestExecutionResult result =
-                            testPluginAgainst(coreCoordinates, plugin, mconfig, pomData, pcth);
-                    if (result.getTestDetails().isSuccess()) {
-                        status = TestStatus.SUCCESS;
-                    } else {
-                        status = TestStatus.TEST_FAILURES;
-                    }
-                    testDetails.addAll(
-                            config.isStoreAll()
-                                    ? result.getTestDetails().getAll()
-                                    : result.getTestDetails().hasFailures()
-                                            ? result.getTestDetails().getFailed()
-                                            : Collections.emptySet());
-                } catch (PomExecutionException e) {
-                    if (!e.succeededPluginArtifactIds.contains("maven-compiler-plugin")) {
-                        status = TestStatus.COMPILATION_ERROR;
-                    } else if (!e.getTestDetails().hasBeenExecuted()) {
-                        // testing was not able to start properly (i.e: invalid exclusion list file
-                        // format)
-                        status = TestStatus.INTERNAL_ERROR;
-                    } else if (e.getTestDetails().hasFailures()) {
-                        status = TestStatus.TEST_FAILURES;
-                    } else { // ???
-                        status = TestStatus.INTERNAL_ERROR;
-                    }
-                    errorMessage = e.getErrorMessage();
-                    testDetails.addAll(
-                            config.isStoreAll()
-                                    ? e.getTestDetails().getAll()
-                                    : e.getTestDetails().hasFailures()
-                                            ? e.getTestDetails().getFailed()
-                                            : Collections.emptySet());
-                } catch (Error e) {
-                    // Rethrow the error ... something is wrong !
-                    throw e;
-                } catch (Throwable t) {
-                    status = TestStatus.INTERNAL_ERROR;
+                PomData pomData = remote.retrievePomData();
+                testPluginAgainst(coreCoordinates, plugin, mconfig, pomData, pcth);
+            } catch (PluginCompatibilityTesterException e) {
+                if (lastException != null) {
+                    e.addSuppressed(lastException);
+                }
+                lastException = e;
+                if (config.isFailFast()) {
+                    break;
+                } else {
```

Comment:
Implements the design described above: in the normal case (a CI or local build specifying one plugin) any failure is immediately fatal. In the case where a local build is passing `-failFast false` and multiple plugins, we temporarily suppress `PluginCompatibilityTesterExeption`s and throw all the ones we encountered after testing all plugins (other errors are still immediately fatal).